engine/src/main/java/org/destinationsol/rubble/systems/RubbleCreationSystem.java

```diff
@@ -39,13 +51,76 @@
  */
 public class RubbleCreationSystem implements EventReceiver {
 
+    public static final float MIN_SCALE = .07f;
+    public static final float MAX_SCALE = .12f;
+    public static final float SIZE_TO_SHARD_COUNT = 13f;
+    private static final float MAX_ROT_SPD = 5f;
+    private static final float MAX_SPD = 40f;
+
     @In
     private RubbleBuilder rubbleBuilder;
 
     @In
     private SolGame solGame;
 
-    //TODO once Shards are entities, this needs to be refactored to replace ShardBuilder
+    @In
+    private EntitySystemManager entitySystemManager;
+
+    /**
+     * This method creates pieces of rubble using values from the object that they are being creating from. It
+     * initializes entities for each piece of rubble with the relevant component
+     * @param pos Position component of parent entity
+     * @param vel Velocity component of parent entity
+     * @param angle Angle component of parent entity
+     * @param size Size component of parent entity, used to determine amount of rubble to generate
+     */
+    public void buildRubblePieces(Position pos, Velocity vel, Angle angle, Size size) {
+        int count = (int) (size.size * SIZE_TO_SHARD_COUNT);
+        Vector2 basePos = pos.position;
+        for (int i = 0; i < count; i++) {
+
+            //Create graphics component
+            RenderableElement element = new RenderableElement();
+            element.texture = SolRandom.randomElement(Assets.listTexturesMatching("engine:shard_.*"));
+            element.drawableLevel = DrawableLevel.PROJECTILES;
+            element.graphicsOffset = new Vector2();
+
+            float scale = SolRandom.randomFloat(MIN_SCALE, MAX_SCALE);
+            element.setSize(scale);
+
+            element.relativePosition = new Vector2();
+            element.tint = Color.WHITE;
+            Renderable graphicsComponent = new Renderable();
+            graphicsComponent.elements.add(element);
+
+            //Create position component
+            float velocityAngle = SolRandom.randomFloat(180);
+            Vector2 position = new Vector2();
+            SolMath.fromAl(position, velocityAngle, SolRandom.randomFloat(size.size));
+            position.add(basePos);
+            Position positionComponent = new Position();
+            positionComponent.position = position;
+
+            //Create health component
+            Health health = new Health();
+            health.currentHealth = 1;
+
+            //Create size component
+            Size sizeComponent = new Size();
+            sizeComponent.size = scale;
+
+            //Create velocity component
+            Velocity velocityComponent = new Velocity();
+            Vector2 velocity = SolMath.fromAl(velocityAngle, SolRandom.randomFloat(MAX_SPD));
+            velocity.add(vel.velocity);
+            SolMath.free(velocity);
```

Comment:
By freeing the velocity before you set it in the component, you're telling the game that it can re-use the vector, which can lead to unspecified behavior. This line should probably be moved to after the entity is created.